src/components/NewsletterSignup/NewsletterSignup.tsx

```diff
@@ -0,0 +1,227 @@
+] as const;
+
+export type NewsletterSignupViewType =
+  typeof newsletterSignupViewTypeArray[number];
+
+interface NewsletterSignupProps {
+  /** Additional class name to add. */
+  className?: string;
+  /** Used to add additional information to the default confirmation message in
+   * the confirmation view. */
+  confirmationText?: string | JSX.Element;
+  /** Used to add description text above the form input fields in
+   * the initial/form view. Accepts a string or a JSX */
+  descriptionText?: string | JSX.Element;
+  /** Optional: Used to populate the Text component rendered below the input field. */
+  formHelperText?: string | JSX.Element;
+  /** A data object containing key/value pairs that will be added to the form
+   * field submitted data. */
+  hiddenFields?: any;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Toggles the invalid state for the email field. */
+  isInvalidEmail?: boolean;
+  /* Optional value to determine the section color highlight */
+  newsletterSignupType?: SectionTypes;
+  /** A callback handler function that will be called when the form is submitted.
+   * The function provided must expect an object of key:value pairs. This object will
+   * include {email}, being the user's input, plus any objects passed via the hiddenValues prop.
+   */
+  onSubmit: (values: { [key: string]: string }) => any;
+  /** Used to populate the `<h3>` header title. */
+  title: string;
+  /** Used to specify what is displayed in the component content area. */
+  view?: NewsletterSignupViewType;
+}
+
+/**
+ * The NewsletterSignup component provides a way for patrons to register for an
+ * email-based newsletter distribution list.
+ *
+ * The component can show four different views, depending on the state of the
+ * email submission: form, loading, confirmation, and error.
+ */
+export const NewsletterSignup = chakra(
+  forwardRef<any, NewsletterSignupProps>(
+    (
+      {
+        className,
+        confirmationText = "Thank you! You have successfully subscribed to our email updates! You can update your email subscription preferences at any time using the links at the bottom of the email.",
+        descriptionText = "Stay connected with the latest research news from NYPL, including information about our events, programs, exhibitions, and collections.",
+        formHelperText,
+        hiddenFields,
+        id,
+        isInvalidEmail = false,
+        newsletterSignupType = "whatsOn",
+        onSubmit,
+        title = "Sign Up for Our Newsletter!",
+        view = "form",
+        ...rest
+      },
+      ref?
+    ) => {
+      const viewType = view;
+      const [buttonClicked, setButtonClicked] = useState(false);
+      const [email, setEmail] = useState("");
+
+      // Hook into NYPL breakpoint
+      const { isLargerThanMobile } = useNYPLBreakpoints();
+      const focusRef = useRef<HTMLDivElement>(); // @todo Is this needed? It is a holdover from FeedbackBox
```

Comment:
This is need to focus on the view when it is updated. Since the component will no longer keep track of its view, this is technically not needed.

If the consuming app is now in full control of the view, it's hard for the component itself to trigger the updated focus. So perhaps we need to revisit this to better plan out how the consuming app will trigger a focus. 

I'm thinking you can pass a new prop like `focusRef` that is a React ref and in the consuming app you can call `focusRef.current.focus()` whenever the `view` prop is updated.  For example, if a fetch request is made in the consuming app and it is successful, you'd update the `view` prop to "confirmation" and then call `focusRef.current.focus()`. 

Another option can be to call `focusRef.current.focus()` whenever `view` is "confirmation" or "error" in the component. You have to stagger the call, or give it enough time, so that the confirmation or error view can render. Otherwise, you may be attempting to focus on an element that doesn't exist. If it already does exist and you focus to it but it doesn't render with the updated view in time, the screenreader may not pick up the changes.

Do both options make sense? We can chat about this tomorrow with @bigfishdesign13

Reply:
This would seem to be solved.